### PR TITLE
techdocs: Add build timestamp to techdocs_metadata.json

### PR DESCRIPTION
**Pull request by OrkoHunter**
Co-authored-by: Raghunandan Balachandran <soapraj@gmail.com>

Needed as a prerequisite to moving from using in-memory storage info. Build info should be part of TechDocs storage. In case of multi-region Backstage deployments, or even using multiple Kubernetes pods, if each instance creates its separate build info in-memory, it will result in duplicate builds per instance. Also if the pod restarts, all the sites will have to be re-built.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4509-ngclbpf88dvkxr3wwtuvxposuvtb9jnt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60265225fcb22d7c4154018d

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4509-ngclbpf88dvkxr3wwtuvxposuvtb9jnt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60265225fcb22d7c4154018d

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4509-ngclbpf88dvkxr3wwtuvxposuvtb9jnt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60265225fcb22d7c4154018d

File: packages/techdocs-common/src/stages/generate/helpers.test.ts [hunk 1]
```diff
@@ -329,4 +331,48 @@ describe('helpers', () => {
...
+      // Check if the file exists
+      await expect(
+        fs.access(filePath, fs.constants.F_OK),
+      ).resolves.not.toThrowError();
+    });
+
+    it('should throw error when the JSON is invalid', async () => {
+      const filePath = path.join(rootDir, 'invalid_techdocs_metadata.json');
+
+      // Check if the file exists
```
Comment (soapraj) [inline packages/techdocs-common/src/stages/generate/helpers.test.ts:363]: nit: looks like copy-pasta from previous test

**Review by soapraj — APPROVED** [review]

File: packages/techdocs-common/src/stages/generate/helpers.test.ts (hunk 1 above)
Comment (OrkoHunter) [reply packages/techdocs-common/src/stages/generate/helpers.test.ts:363]: Yes! 😁

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4509-ngclbpf88dvkxr3wwtuvxposuvtb9jnt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60265225fcb22d7c4154018d

**Pull request merged**